examples/distributed/matmul.py

```diff
@@ -0,0 +1,50 @@
+import numpy
+import cupy
+from cupyx.distributed.array import distributed_array
+from cupyx.distributed.array import make_2d_index_map
+
+
+N = 300
+M = 400
+P = 200
+
+data_a = numpy.arange(N * M).reshape(N, M)
+data_b = numpy.arange(M * P).reshape(M, P)
+
+#     0    M/3    M
+#   0 +-----+-----+
+#     | 0 1 |  2  |
+# N/2 +-----+-----+
+#     | 0 3 | 1 2 |
+#   N +-----+-----+
```

Comment:
Q: Does this sample require 4 GPUs?

Reply:
Yeah, we need to add a runtime check here! great catch :)